src/Workspaces/Core/Portable/Options/OptionServiceFactory.cs

```diff
@@ -121,7 +121,9 @@ private ImmutableArray<EventHandler<OptionChangedEventArgs>> GetEventHandlers()
             public SerializableOptionSet GetSerializableOptionsSnapshot(ImmutableHashSet<string> languages) => _globalOptionService.GetSerializableOptionsSnapshot(languages, this);
             public object? GetOption(OptionKey optionKey) => _globalOptionService.GetOption(optionKey);
             [return: MaybeNull] public T GetOption<T>(Option<T> option) => _globalOptionService.GetOption(option);
+            [return: MaybeNull] public T GetOption<T>(Option2<T> option) => _globalOptionService.GetOption(option);
             [return: MaybeNull] public T GetOption<T>(PerLanguageOption<T> option, string? languageName) => _globalOptionService.GetOption(option, languageName);
+            [return: MaybeNull] public T GetOption<T>(PerLanguageOption2<T> option, string? languageName) => _globalOptionService.GetOption(option, languageName);
```

Comment:
📝 All of these methods have the wrong return type annotation.

Reply:
Yep, will track it as a follow-up item

Reply:
Tracked in last bullet item in https://github.com/dotnet/roslyn/issues/41462